test/autorest/lrogroup/lros_test.go

```diff
@@ -69,1142 +75,974 @@ func TestLROBeginDelete202NoRetry204(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginPut200Succeeded(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	_, err = poller.ResumeToken()
+	if err == nil {
+		t.Fatal("Expected an error but did not receive one")
+	}
+	res, err := resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, res.RawResponse, 200)
+	helpers.DeepEqualOrFatal(t, res.Product, &lrogroup.Product{
+		Resource: lrogroup.Resource{
+			ID:   to.StringPtr("100"),
+			Name: to.StringPtr("foo"),
+		},
+	})
+}
+
+func TestLROBeginPut200SucceededNoState(t *testing.T) {
+	t.Skip("problem with the poller")
+	op := getLROSOperations(t)
+	resp, err := op.BeginPut200SucceededNoState(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	_, err = poller.ResumeToken()
+	if err == nil {
+		t.Fatal("Expected an error but did not receive one")
+	}
+	res, err := resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, res.RawResponse, 200)
+	helpers.DeepEqualOrFatal(t, res.Product, &lrogroup.Product{
+		Resource: lrogroup.Resource{
+			ID:   to.StringPtr("100"),
+			Name: to.StringPtr("foo"),
+		},
+	})
+}
+
+// TODO check if this test should actually be returning a 200 or a 204
+func TestLROBeginPut200UpdatingSucceeded204(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginPut200UpdatingSucceeded204(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumePut200UpdatingSucceeded204(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res, err := resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, res.RawResponse, 200)
+	helpers.DeepEqualOrFatal(t, res.Product, &lrogroup.Product{
+		Resource: lrogroup.Resource{
+			ID:   to.StringPtr("100"),
+			Name: to.StringPtr("foo"),
+		},
+		Properties: &lrogroup.ProductProperties{
+			ProvisioningState: to.StringPtr("Succeeded"),
+		},
+	})
+}
+
+func TestLROBeginPut201CreatingFailed200(t *testing.T) {
+	t.Skip("missing error info message returned for error")
+	op := getLROSOperations(t)
+	resp, err := op.BeginPut201CreatingFailed200(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumePut201CreatingFailed200(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = resp.PollUntilDone(context.Background(), 1*time.Millisecond)
+	if err == nil {
+		t.Fatal("expected an error but did not receive one")
+	}
+}
+
+func TestLROBeginPut201CreatingSucceeded200(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginPut201CreatingSucceeded200(context.Background(), nil)
```

Comment:
It's interesting that none of these REST endpoints fail without a payload.  Might be good to add one for some of the APIs (in track 1 we just specified an empty `lrogroup.Product{}`).

Reply:
Done I added it for a few of the tests.

Reply:
And yes it's strange that they dont fail without a payload